modules/core/src/test/scala-3/com/siriusxm/snapshot4s/ReprSpec.scala

```diff
@@ -103,6 +105,64 @@ object ReprSpec extends FunSuite:
     "Could not find implicit instance for Repr[snapshot4s.ReprSpec.RecursiveProduct]"
   )
 
+  test("Repr respects custom Repr instances for fields") {
+    case class WithString(name: String)
+
+    // Use Scala 3 given syntax
+    given customStringRepr: Repr[String] = _ => "CustomString(test)"
+    val repr                             = summon[Repr[WithString]]
+
+    val input = WithString("test")
+    expect.same("WithString(name = CustomString(test))", repr.toSourceString(input))
+  }
+
+  test("Repr respects custom Repr instances for lists") {
```

Comment:
Do you think these tests need to be specific to Scala 3 syntax?

They're testing the resolution of `Repr` for `List` and other datatypes, but we don't expect this to be different between Scala versions.

It would be more concise to have most of these tests in the shared `ReprTestCases` using the `implicit` keyword, and have a single Scala-3 specific test covering the `given` syntax.